lib/model/folder.go

```diff
@@ -450,6 +451,51 @@ func (f *folder) scanSubdirs(subDirs []string) error {
 
 		batch.append(res.File)
 		changes++
+
+		err := snap.WithPathsMatchingBlocksHash(res.File.BlocksHash, func(path string) bool {
+			select {
+			case <-f.ctx.Done():
+				return false
+			default:
+			}
+
+			cf, ok := snap.Get(protocol.LocalDeviceID, path)
+			if !ok {
+				return true
+			}
+			if !bytes.Equal(cf.BlocksHash, res.File.BlocksHash) {
+				l.Debugf("Mismatching block map list hashes: got %x expected %x", cf.BlocksHash, res.File.BlocksHash)
```

Comment:
That's should never happen though, right? I'd drop this check or move it to the db package and then correct the inconsistency (remove key) and log at a higher level.

Reply:
This potentially means we're somehow out of sync. 
I don't think I can control that easily in the db package, as I'd have to retrieve every key I find to verify this.

Reply:
Right, actually now that you pointed it out: The first thing you do here is get the actual fileinfo - why don't you do that in the db code? Then you also don't need to introduce the new string iterator.

Reply:
That's a fair suggestion, I could also deal with flag filtering and "fixing up" there.

Reply:
Actually, this is all in a read-only transaction, so I don't think I can do any fixing up there.